lib/mixins/listeners-mixin.html

```diff
@@ -0,0 +1,68 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="template-stamp.html">
+<link rel="import" href="../utils/mixin.html">
+<script>
+  (function() {
+    'use strict';
+
+    /**
+     * Element class mixin to declaratively create event listeners on the element,
+     * similar to the "listeners" configuration from Polymer 1.x
+     *
+     * By returning a map from a static class property named "listeners",
+     * the element will create event listeners that handle events with functions defined on the class
+     *
+     * For example, if an element implements "listeners" like so, an instance of that element will have
+     * a "foo" event handler that calls "fooHandler":
+     *
+     * ```
+     * class Foo extends Polymer.Listeners(Polymer.Element) {
+     *   static get listeners() {
+     *     "foo": "fooHandler"
+     *   }
+     *   fooHandler(ev) {...}
+     * }
+     * ```
+     *
+     * @polymerMixin
+     * @mixes Polymer.TemplateStamp
+     * @memberof Polymer
+     * @summary Element class mixin to declaratively create event listeners on the element
+     */
+    Polymer.Listeners = Polymer.dedupingMixin((base) => {
+      /**
+       * @constructor
+       * @extends {base}
+       * @implements {Polymer_TemplateStamp}
+       */
+      let listenerBase = Polymer.TemplateStamp(base);
+
+      /**
+       * @polymerMixinClass
+       * @extends {listenerBase}
+       * @implements {Polymer_Listeners}
+       */
+      class Listeners extends listenerBase {
+        ready() {
+          let listeners = this.constructor.listeners;
```

Comment:
Subclassing is tricky with this implementation. Ideally you can easily inherit/override these.